src/config/__init__.py

```diff
@@ -46,9 +51,6 @@ def get_value(
             value.set_file(user_config_filename, base_for_paths=True)
             break
 
-    # read configuration from environment variables
-    value.set_env(sep="__")
-
```

Comment:
This is ok to remove because we're not nesting configuration (i.e. none of the configuration variables have type dictionary)

Reply:
removing this breaks a bunch of tests from config

Reply:
i'm rolling this back because this seems very complex to fix and we shouldn't need this if we set the env variables correctly

Reply:
Actually, these needs to be restored because this line does use nesting:

https://github.com/formiclabs/deeporigin-client/blob/7a4897a2a9b7c18abdc06ce441afa90931aa0535/src/config/__init__.py#L71-L77

These tests highlight this

https://github.com/formiclabs/deeporigin-client/blob/7a4897a2a9b7c18abdc06ce441afa90931aa0535/tests/test_config.py#L64

https://github.com/formiclabs/deeporigin-client/blob/7a4897a2a9b7c18abdc06ce441afa90931aa0535/tests/test_config.py#L71

Reply:
Why are you trying to remove this?

This simply controls the separator between levels of configuration.

Example (`__`) in `DEEP_ORIGIN_FEATURE_FLAGS__VARIABLES`.
- `DEEP_ORIGIN`: prefix defined at https://github.com/formiclabs/deeporigin-client/blob/7a4897a2a9b7c18abdc06ce441afa90931aa0535/src/config/__init__.py#L37
- `_`: prefix separator. I believe this is defined by confuse, and we can't configure it.
- `FEATURE_FLAGS`: first level config defined at https://github.com/formiclabs/deeporigin-client/blob/7a4897a2a9b7c18abdc06ce441afa90931aa0535/src/config/__init__.py#L71
- `__`: level separator defined at https://github.com/formiclabs/deeporigin-client/blob/7a4897a2a9b7c18abdc06ce441afa90931aa0535/src/config/__init__.py#L50
- `VARIABLES`: second level config defined at https://github.com/formiclabs/deeporigin-client/blob/7a4897a2a9b7c18abdc06ce441afa90931aa0535/src/config/__init__.py#L74

Reply:
yes...and i think there is a simpler workaround for the underlying issue